clang/lib/Basic/Targets/RISCV.h

```diff
@@ -181,11 +187,17 @@ class RISCVTargetInfo : public TargetInfo {
   bool hasExtIntType() const override { return true; }
 
   CallingConvCheckResult checkCallingConvention(CallingConv CC) const override {
-    if ((CC == CallingConv::CC_CHERICCall) ||
-        (CC == CallingConv::CC_CHERICCallee) ||
+    if (CC == CallingConv::CC_CHERICCall) {
+      // NB: suppress warning when cheriot-baremetal
+      return ABI == "cheriot" || ABI == "cheriot-baremetal" ? CCCR_OK : CCCR_Warning;
+    }
+    if ((CC == CallingConv::CC_CHERICCallee) ||
         (CC == CallingConv::CC_CHERICCallback) ||
-        (CC == CallingConv::CC_CHERILibCall))
+        (CC == CallingConv::CC_CHERILibCall)) {
+      // NB: with cheriot-baremetal the caller will generate a warning
+      //   and downgrade to the default target CC
```

Comment:
Not sure I understand the need for this change; you never generate a CC_CHERICCall, right, for baremetal?